src/GovUk.Education.ExploreEducationStatistics.Public.Data.Model/DataSetVersionImportStage.cs

```diff
@@ -7,5 +7,26 @@ public enum DataSetVersionImportStage
 {
     Pending,
     CopyingCsvFiles,
+    ImportingMetadata,
+    ImportingData,
+    WritingDataFiles,
     Completing
 }
+
+public static class DataSetVersionImportStageExtensions
+{
+    public static DataSetVersionImportStage PreviousStage(this DataSetVersionImportStage stage)
+    {
+        return stage switch
+        {
+            DataSetVersionImportStage.CopyingCsvFiles => DataSetVersionImportStage.Pending,
+            DataSetVersionImportStage.ImportingMetadata => DataSetVersionImportStage.CopyingCsvFiles,
+            DataSetVersionImportStage.ImportingData => DataSetVersionImportStage.ImportingMetadata,
+            DataSetVersionImportStage.WritingDataFiles => DataSetVersionImportStage.ImportingData,
+            DataSetVersionImportStage.Completing => DataSetVersionImportStage.WritingDataFiles,
+            _ => throw new ArgumentOutOfRangeException(nameof(stage),
+                stage,
+                $"No previous stage has been defined for '{stage}'")
+        };
```

Comment:
Cool, I did think about using the ordinal position before but I opted for this explicit approach mainly because it was easier to write! 😆 

Thanks for taking the time to work that out, I've swapped it out with your suggestion now. 🙂